src/MSBuild.Abstractions/MSBuildHelpers.cs

```diff
@@ -217,9 +217,10 @@ public static bool HasWPFOrWinForms(ProjectPropertyGroupElement propGroup)
         /// <summary>
         /// Checks if a given TFM is not .NET Framework.
         /// </summary>
-        public static bool IsNotNetFramework(string tfm) =>
-            !tfm.ContainsIgnoreCase(MSBuildFacts.NetcoreappPrelude)
-            && !tfm.ContainsIgnoreCase(MSBuildFacts.NetstandardPrelude);
+        public static bool IsNotNetFramework(string? tfm) =>
+            tfm is null
```

Comment:
I would not allow this to accept `null`. I don't think this check should succeed becasue we passed in null.